01_Exercises/Lab 4 to 6/main-family-expenses-app.py

```diff
@@ -0,0 +1,338 @@
+import removeExpenses
+import findExpenses
+import repository
+import interface
+import undoRedo
+import reports
+import os
+
+
+def collect_data():
+    """It collects the data and stores it as object"""
+    expense_date = input("\nDate (dd-mm-yy): ")
+    expense_amount = input("Amount: ")
+    expense_type = input("Type (Food, Bills, Clothes, Phone or Other): ")
+    return repository.Expense(expense_date, expense_amount, expense_type)
+
+
+def main():
+    all_expenses = []
+    undo = []
+
+    while True:
+        os.system("cls")
+        user_interface = interface.UI()
+        user_interface.main_menu()
+        options = interface.choose_option()
+
+        if options == "1":  # add expenses
+            while True:
+                os.system("cls")
+                print("Please write your expense")
+                interface.display_expense(all_expenses)
+
+                data_collected = collect_data()
+                data_collected_details = data_collected.get_all_expense_details()
+                display_data = " ".join(data_collected_details)
+
+                if repository.check_all_data(data_collected):
+                    all_expenses.append(data_collected_details)
```

Comment:
This should behaviour sohuld be separated in a Repository class which controller should interact with